src/main/java/org/sonarqube/gradle/SonarResolverTask.java

```diff
@@ -124,4 +134,26 @@ void run() throws IOException {
       LOGGER.info("Resolved properties for " + displayName + " and wrote them to " + getOutputFile() + ".");
     }
   }
+
+  // Suppress warning about using deprecated API for Gradle < 7 compatibility
+  @SuppressWarnings("java:S1874")
+  private FileCollection getClasspathFromSourceSets(String sourceSetName) {
+    if(isAtLeastGradle7()){
+      JavaPluginExtension javaExt = getExtensions().findByType(JavaPluginExtension.class);
+      if(javaExt != null){
+        return SonarUtils.getClassPathFromSourceSets(sourceSetName, javaExt.getSourceSets());
+      }
```

Comment:
We can maybe try to increase the coverage here by adding a groovy test with a project that has the java plugin